src/to_vcf.py

```diff
@@ -34,7 +36,50 @@ def parse_args():
     args = parser.parse_args()
     return args
 
-def write_to_vcf(fn_chain: str, fn_vcf: str, fn_targetfasta: str, fn_queryfasta: str):
+
+def write_to_vcf(
+    f: TextIO, targetref: pysam.FastaFile=None, queryref: pysam.FastaFile=None
+) -> str:
+    for line in f:
+        fields = line.split()
+        if len(fields) == 0:
+            pass
+        elif fields[0] == 'chain':
+            c = utils.Chain(fields)
+        else:
+            c.add_record(fields)
+            if len(fields) == 1:
+                yield c.to_vcf(targetref=targetref, queryref=queryref)
+                c = None
+
+
+def write_to_vcf_io(
+    fn_chain: str, fn_vcf: str,
+    fn_targetfasta: str='', fn_queryfasta: str=''
+) -> None:
+    if fn_chain == '-':
+        f = sys.stdin
+    else:
+        f = open(fn_chain, 'r')
+    if fn_vcf:
+        fo = open(fn_vcf, 'w')
+    else:
+        fo = sys.stdout
+
+    targetref = utils.fasta_reader(fn_targetfasta)
+    queryref = utils.fasta_reader(fn_queryfasta)
+
+    print(utils.vcf_header(utils.get_target_entries(fn_chain)), file=fo)
+
+    out = write_to_vcf(f=f, targetref=targetref, queryref=queryref)
+    while True:
+        try:
+            print(next(out), file=fo)
+        except StopIteration:
+            break
+
+
+def write_to_vcf2(fn_chain: str, fn_vcf: str, fn_targetfasta: str, fn_queryfasta: str):
```

Comment:
Can get rid of write_to_vcf2, no?

Reply:
I just removed it. Thanks for pointing this out!